Framework/Algorithms/src/SofQCommon.cpp

```diff
@@ -167,20 +205,15 @@ SofQCommon::qBinHintsIndirect(const API::MatrixWorkspace &ws, const double minE,
   using namespace Mantid::PhysicalConstants;
   auto minQ = std::numeric_limits<double>::max();
   auto maxQ = std::numeric_limits<double>::lowest();
-  const auto &detectorInfo = ws.detectorInfo();
-  for (size_t i = 0; i < detectorInfo.size(); ++i) {
-    if (detectorInfo.isMasked(i) || detectorInfo.isMonitor(i)) {
+  const auto &spectrumInfo = ws.spectrumInfo();
+  for (size_t i = 0; i < spectrumInfo.size(); ++i) {
```

Comment:
Are you sure about this change? I see you adjusted the doxygen above. One thing to double check: What does `getEFixed` to for a `DetectorGroup`? Unless it returns an average (of the E_f looked up in the parameter map) for all contained detectors I think you cannot change this here.